aspnetcore/host-and-deploy/docker/building-net-docker-images.md

```diff
@@ -0,0 +1,211 @@
+  * [macOS](https://docs.docker.com/docker-for-mac/install/)
+  * [Windows](https://docs.docker.com/docker-for-windows/install/)
+
+* [git](https://git-scm.com/download)
+
+## Download the sample application
+
+* Download the sample by cloning the [.NET Core Docker repository](https://github.com/dotnet/dotnet-docker): 
+
+  ```console
+  git clone https://github.com/dotnet/dotnet-docker
+  ```
+
+## Run the app locally
+
+* Navigate to the project folder at *dotnet-docker/samples/aspnetapp/aspnetapp*.
+
+* Run the following command to build and run the app locally:
+
+  ```console
+  dotnet run
+  ```
+
+* After the application starts, visit `http://localhost:5000` in a browser to verify that the app is working.
+
+* Press Ctrl+C at the command prompt to stop the app.
+
+## Run in a Linux container
+
+* In the Docker client, switch to Linux containers.
+
+* Navigate to the Dockerfile folder at *dotnet-docker/samples/aspnetapp*.
+
+* Run the following commands to build and run the sample in Docker:
+
+  ```console
+  docker build -t aspnetapp .
+  docker run -it --rm -p 5000:80 --name aspnetcore_sample aspnetapp
+  ```
+
+  The `build` command arguments:
+  * Name the image aspnetapp.
+  * Look for the Dockerfile in the current folder (the period at the end).
+
+  The run command arguments:
+  * Allocate a pseudo-TTY and keep it open even if not attached. (Same effect as `--interactive --tty`.)
+  * Automatically remove the container when it exits.
+  * Map port 5000 on the local machine to port 80 in the container.
+  * Name the container aspnetcore_sample.
+  * Specify the aspnetapp image.
+
+* After the application starts, see the home page at `http://localhost:5000` in your web browser.
+
+## Run in a Windows container
+
+* In the Docker client, switch to Windows containers.
+
+Navigate to the docker file folder at `dotnet-docker/samples/aspnetapp`.
+
+* Run the following commands to build and run the sample in Docker:
+
+  ```console
+  docker build -t aspnetapp .
+  docker run -it --rm --name aspnetcore_sample aspnetapp
+  ```
+
+* For Windows containers, you need the IP address of the container (browsing to `http://localhost:5000` won't work):
+  * Open up another command prompt.
+  * Run `docker ps` to see the running containers. Verify that the "aspnetcore_sample" container is there.
+  * Run `docker exec aspnetcore_sample ipconfig` to display the IP address of the container. The output from the command looks like this example:
+
+    ```console
+    Ethernet adapter Ethernet:
+
+       Connection-specific DNS Suffix  . : contoso.com
+       Link-local IPv6 Address . . . . . : fe80::1967:6598:124:cfa3%4
+       IPv4 Address. . . . . . . . . . . : 172.29.245.43
+       Subnet Mask . . . . . . . . . . . : 255.255.240.0
+       Default Gateway . . . . . . . . . : 172.29.240.1
+    ```
+
+* Copy the container IPv4 address (for example, 172.29.245.43) and paste into the browser address bar to see the sample app's home page.
```

Comment:
Copy the container IPv4 address (for example, 172.29.245.43) and paste into the browser address bar to test the app.